tests/lib/rules/no-invalid-this.js

```diff
@@ -83,7 +83,7 @@ function extractPatterns(patterns, type) {
     }));
 
     // Flatten.
-    return Array.prototype.concat.apply([], patternsList);
+    return [].concat(...patternsList);
```

Comment:
possibly be `return [...patternsList)]`?

Reply:
Actually, I tried this and it doesn't work. `patternList` is an array of arrays that needs to be flattened. `[...patternList]` just shallow-copies the array of arrays. `[].concat(...patternsList)` flattens the array. Thanks for the suggestion though, it was worth a try!